src/bin/compress.rs

```diff
@@ -76,43 +79,58 @@ fn main() {
     
     if args.shuffle {
-        programs.shuffle(&mut rand::thread_rng());
+        train_programs.shuffle(&mut rand::thread_rng());
     }
     if let Some(n) = args.truncate {
-        programs.truncate(n);
+        train_programs.truncate(n);
     }
     
     // parse the program strings into expressions
-    let programs: Vec<Expr> = programs.iter().map(|p| p.parse().unwrap()).collect();
+    let train_programs: Vec<Expr> = train_programs.iter().map(|p| p.parse().unwrap()).collect();
+    let test_programs: Option<Vec<Expr>> = if let Some(ps) = test_programs { Some(ps.iter().map(|p| p.parse().unwrap()).collect()) } else { None };
+    
 
     // for prog in programs.iter() {
     //     println!("{}", prog);
     // }
     println!("{}","**********".blue().bold());
     println!("{}","* Stitch *".blue().bold());
     println!("{}","**********".blue().bold());
-    programs_info(&programs);
+    println!("{}","* Training programs: *".blue().bold());
+    programs_info(&train_programs);
+    println!("{}","* Testing programs: *".blue().bold());
+    if let Some(test_ps) = &test_programs {
+        programs_info(test_ps);
+    } else {
+        println!{"No test programs given"}
+    }
 
     // build a single `Expr::Programs` node from these programs. Stitch uses these because often we want to treat
     // different parts of the same programs the same way that we treat different parts of different programs, so
     // treating everything as one big expression makes sense.
-    let programs: Expr = Expr::programs(programs);
+    let train_programs: Expr = Expr::programs(train_programs);
+    let test_programs: Option<Expr> = if let Some(ps) = test_programs { Some(Expr::programs(ps)) } else { None };
 
-    if programs.to_string_curried(None).contains("(app (lam") {
+    if train_programs.to_string_curried(None).contains("(app (lam")  {
         println!("Normal dreamcoder programs never have unapplied lambdas in them! Who knows what might happen if you run this. Probably it will be fine");
     }
 
-    let step_results = compression(&programs, args.iterations, &args.step, &tasks, num_prior_inventions);
+    let step_results = compression(&train_programs, &test_programs, args.iterations, &args.step, &tasks, num_prior_inventions);
 
     // write everything to json
     let out = json!({
         "cmd": std::env::args().join(" "),
         "args": args,
-        "original_cost": programs.cost(),
-        "original": programs.split_programs().iter().map(|p| p.to_string()).collect::<Vec<String>>(),
+        "train_original_cost": train_programs.cost(),
+        "train_original": train_programs.split_programs().iter().map(|p| p.to_string()).collect::<Vec<String>>(),
+        "test_original_cost": if let Some(ps) = &test_programs { ps.cost() } else { -1 },
+        "test_original": if let Some(ps) = &test_programs { ps.split_programs().iter().map(|p| p.to_string()).collect::<Vec<String>>() } else { vec![] },
```

Comment:
See comment for line 96.

Reply:
`map_or` with json `Value::Νull` as described above. For both of these lines. Lets stick with nulls for json values that dont make sense like this.